openkfw/TruBudget
# #1658 api: JWT with RS256

### Checklist




Some other services also use JWT_SECRET in certain scenarios. Check if there are further changes needed.
- [x] email-notification-service (doesn't work with authentication turned on, see https://github.com/openkfw/TruBudget/issues/1769
- [x] excel-export-service
- [ ] blockchain (connects to email service) - not working on m2 machine, can't fix

### Description

### How to test

1. Set new env vars: JWT_ALGORITHM=RS256, JWT_SECRET=base 64 pem encoded private key, JWT_PUBLIC_KEY=base 64 pem encoded public key
2. log in
3. token cookie should be RS256
4. do some stuff
5. old configuration should work too (no JWT_ALGORITH, or set to HS256, no public key, whatever secret)




Closes #1658
